tests/integration/istio-injection/istioinjection/main_test.go

```diff
@@ -0,0 +1,74 @@
+package istioinjection
+
+import (
+	"fmt"
+	"testing"
+
+	log "github.com/sirupsen/logrus"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	namespaceNameRoot = "istio-injection-tests"
+	noSidecar         = 1
+	hasSidecar        = 2
+)
+
+type TestSuite struct {
+	k8sClient *kubernetes.Clientset
+	namespace string
+}
+
+func TestIstioInjection(t *testing.T) {
+	namespace := fmt.Sprintf("%s-%s", namespaceNameRoot, generateRandomString(8))
+	if namespace == "" {
+		log.Fatal("Namespace not set.")
+	}
+
+	testSuite := &TestSuite{namespace: namespace}
+	testSuite.initK8sClient()
+
+	defer testSuite.deleteNamespace()
+	if err := testSuite.createNamespace(); err != nil {
```

Comment:
maybe it's just me, but creating a namespace per test case seems like a better idea. The `createNamespace` func could accept a bool to control the injection. Ultimately, one test case would cover the transition scenario and test the injection before and after the update.